lookout/style/format/utils.py

```diff
@@ -0,0 +1,28 @@
+from copy import deepcopy
+from typing import Mapping
+
+
+def merge_dicts(dict1: Mapping, dict2: Mapping) -> Mapping:
+    """
+    Deep merge of nested dictionaries.
+    Example:
+    >>> a = {1: 1, 2: {3: 3, 4: 4}}
+    >>> b = {1: 10, 2: {3: 30, 4: 4, 5: 5}}
+    >>> merge_dicts(a, b)
+    {1: 10, 2: {3: 30, 4: 4, 5: 5}}
+    >>> merge_dicts(b, a)
+    {1: 1, 2: {3: 3, 4: 4, 5: 5}}
+
+    :return: new merged dictionary
+    """
+
+    res = deepcopy(dict1)
+    stack = [(res, dict2)]
+    while stack:
```

Comment:
This implementation allows for
```python
def merge_dicts(*dicts: Mapping)
```

Reply:
it is not commutative, so I prefer to leave as is. 
`merge_dicts(a,b,c) != merge_dicts(a,c,b)`
it can lead to bugs.

Reply:
if you are fine with that of course.

Reply:
That's fine (btw please include this into the docstring!). The current version is equally prone to the same bugs, also e.g. `merge_dicts(merge_dicts(a, merge_dicts(b, c)), d)` is completely non-readable.

Reply:
agree

Reply:
So?.. 😄

Reply:
@zurk

Reply:
https://github.com/src-d/style-analyzer/pull/200#issuecomment-428284931